Hiding the `Patient Home` button when in Patient Home page

GSMPRANEETH opened the pull request:
## Proposed Changes

Fixes #14009 
- Hiding the button when the user is already in page that has `patients/verify` in the url

<img width="1920" height="1080" alt="image_2025-10-14_214601297" src="https://github.com/user-attachments/assets/7ed63859-c0e6-45f0-99c7-78cf6cd28b43" />


Tagging: @ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [x] Add specs that demonstrate the bug or test the new feature.
- [x] Update [product documentation](https://docs.ohc.network).
- [x] Ensure that UI text is placed in I18n files.
- [x] Prepare a screenshot or demo video for the changelog entry and attach it to the issue.
- [x] Request peer reviews.
- [x] Complete QA on mobile devices.
- [x] Complete QA on desktop devices.


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Patient info hover card now detects when you're already on a patient's verification page and hides the redundant primary action, simplifying available actions and preventing duplicate navigation options.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Commit:
using hidden when link has patients/verify

AdityaJ2305 commented on src/components/Patient/PatientInfoHoverCard.tsx:
```diff
@@ -36,7 +36,12 @@ export const PatientInfoHoverCard = ({
+          variant="outline"
+          className="text-gray-950"
+          asChild
+          hidden={location.pathname.includes("/patients/verify")}
```
![Image](https://github.com/user-attachments/assets/9ec668d4-a055-4cc1-9c0b-c0394b67e37f)

It still renders in dom

GSMPRANEETH replied:
Only rendering it when not in the patients home page 👍

yash-learner reviewed (changes requested):
Use raviger's `usePath`  which is common in the codebase or raviger's `useRoute`   instead of `location.pathname`

and 

Also as Aditya mentioned let's not even render that https://github.com/ohcnetwork/care_fe/pull/14053/files#r2431692181

Commit:
Using raviger's usePath and not rendering in DOM when already in page

Commit:
using complete url of patient to safeguard

AdityaJ2305 commented on src/components/Patient/PatientInfoHoverCard.tsx:
```diff
@@ -36,20 +36,29 @@ export const PatientInfoHoverCard = ({
-            {t("patient_home")}
-          </Link>
-        </Button>
+        {!usePath()?.includes(
```
Extract the path out in variable called path and use that instead. It would more cleaner 

`const path = usePath();`

GSMPRANEETH replied:
Using only `"patients/verify"` so wont be needed 😄

AdityaJ2305 replied:
@GSMPRANEETH Didn't get what you want to say ?

yash-learner replied:
@GSMPRANEETH https://github.com/ohcnetwork/care_fe/pull/14053#discussion_r2435506370,  https://github.com/ohcnetwork/care_fe/pull/14053#discussion_r2435590598

and 

why did you preferred `includes`, instead of `endsWith` ?  https://github.com/ohcnetwork/care_fe/pull/14053/commits/a75128759e60a3f668e1363387c68480b0684874

GSMPRANEETH replied:
@AdityaJ2305 
as now only using a short url `patients/verify`, i think there is no need to explicitly define a var for it
@yash-learner 
I was in the fog on the working of `endsWith` 😅 , thought only verifies the end of an url, but the patients home page has var  params in the end, so I prefered `includes`

yash-learner replied:
@GSMPRANEETH 

> I was in the fog on the working of endsWith 😅 , thought only verifies the end of an url, but the patients home page has var params in the end, so I prefered includes

`usePath()` from Raviger gets **only the path** not params from the `URL` -> If you want you can test it in your local by doing a console.log of path via `const path = usePath()`

---

> as now only using a short url patients/verify, i think there is no need to explicitly define a var for it

It still violated rules of hook as pointed by copilot,  https://github.com/ohcnetwork/care_fe/pull/14053#discussion_r2435506370

---

Can you make above changes ?

AdityaJ2305 commented on src/components/Patient/PatientInfoHoverCard.tsx:
```diff
@@ -36,20 +36,29 @@ export const PatientInfoHoverCard = ({
+            year_of_birth: patient.year_of_birth.toString(),
+            partial_id: patient.id.slice(0, 5),
+          }).toString()}`,
+        ) && (
```
Hmm, kind of agree with @copilot here , query param could change and doesn't guarantee every time it would match 

so let's just go ahead with
`/facility/${facilityId}/patients/verify`


@yash-learner any comments on it ? 


PS: Honestly, I’m not a big fan of using the path to hide the button. I’d rather use an optional prop and conditionally render it. But after checking the code, doing that would cause prop drilling since there are too many parent levels involved.

yash-learner replied:
@AdityaJ2305 @GSMPRANEETH 

I think just this is enough and will work

```ts
!path?.endsWith("/patients/verify")
```

----

```ts
        {!path?.endsWith("/patients/verify") && (
          <Button variant="outline" className="text-gray-950" asChild>
            <Link
              basePath="/"
              href={`/facility/${facilityId}/patients/verify?${new URLSearchParams(
                {
                  phone_number: patient.phone_number,
                  year_of_birth: patient.year_of_birth.toString(),
                  partial_id: patient.id.slice(0, 5),
                },
              ).toString()}`}
            >
              {t("patient_home")}
            </Link>
          </Button>
        )}
  ```

GSMPRANEETH replied:
👍

Commit:
using short url

yash-learner committed:
Merge branch 'develop' into issues/14009/patient_home_button

Commit:
complies rules of hook & linter rules

yash-learner reviewed (approved):
Code looks good to me

yash-learner committed:
Merge branch 'develop' into issues/14009/patient_home_button

AdityaJ2305 commented on src/components/Patient/PatientInfoHoverCard.tsx:
```diff
@@ -19,6 +19,9 @@ export const PatientInfoHoverCard = ({
   const { t } = useTranslation();
 
+  const path = usePath();
+  const isHome = path?.endsWith(`/facility/${facilityId}/patients/verify`);
```
```suggestion
  const isPatientHomePage = path?.endsWith(`/facility/${facilityId}/patients/verify`);
```

GSMPRANEETH committed:
Update src/components/Patient/PatientInfoHoverCard.tsx

renaming the var that verifies if user is in patients homepage to `isPatientHomePage` from `isHome`

Co-authored-by: Aditya Jindal <adityajindal2305@gmail.com>

AdityaJ2305 commented:
@GSMPRANEETH Format the code which is causing build fail

Commit:
fixing var name error

AdityaJ2305 reviewed (approved):
LGTM

GSMPRANEETH committed:
Merge branch 'develop' into issues/14009/patient_home_button

GSMPRANEETH commented on src/components/Patient/PatientInfoHoverCard.tsx:
```diff
@@ -19,6 +19,11 @@ export const PatientInfoHoverCard = ({
   const { t } = useTranslation();
 
+  const path = usePath();
+  const isPatientHomePage = path?.endsWith(
+    `/facility/${facilityId}/patients/verify`,
+  );
```
not gonna use includes

nihal467 committed:
Merge branch 'develop' into issues/14009/patient_home_button
